mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrix.scala

```diff
@@ -459,14 +464,155 @@ class BlockMatrix @Since("1.3.0") (
    */
   @Since("1.3.0")
   def multiply(other: BlockMatrix): BlockMatrix = {
+    multiply(other, 1)
+  }
+
+  /**
+   * Left multiplies this [[BlockMatrix]] to `other`, another [[BlockMatrix]]. This method add
```

Comment:
Ok, I feel this is very verbose in terms of documentation. Can we summarize it somehow?
```scala
/**
 * Left multiplies this [[BlockMatrix]] to `other`, another [[BlockMatrix]]. The `colsPerBlock`
 * of this matrix must equal the `rowsPerBlock` of `other`. If `other` contains
 * `SparseMatrix`, they will have to be converted to a `DenseMatrix`. The output
 * [[BlockMatrix]] will only consist of blocks of `DenseMatrix`. This may cause
 * some performance issues until support for multiplying two sparse matrices is added.
 * Blocks with duplicate indices will be added with each other.
 *
 * @param other Matrix `B` in `A * B = C`
 * @param numMidDimSplits Number of splits to cut on the middle dimension when doing multiplication.
 For example, when multiplying a Matrix `A` of size `m x n` with Matrix `B` of size `n x k`, this parameter
 configures the parallelism to use when grouping the matrices. The parallelism will increase from `m x k` to
 `m x k x numMidDimSplits`, which in some cases also reduces total shuffled data.
```

Wrote out a sketch, please put in proper format, i.e. omitted `*` on the last lines